### PR TITLE
Update .NET to version `6.0.100-preview.2.21155.3`

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Update the .NET SDK to .NET 6 preview 2.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/System.Management.Automation/engine/hostifaces/PowerShell.cs [hunk 1]
```diff
@@ -5316,7 +5316,7 @@ private void RemoveFromRemoteRunspaceRunningList()
         private RemoteRunspacePoolInternal GetRemoteRunspacePoolInternal()
         {
             RunspacePool runspacePool = _rsConnection as RunspacePool;
-            return (runspacePool != null) ? (runspacePool.RemoteRunspacePoolInternal) : null;
+            return runspacePool?.RemoteRunspacePoolInternal;
```
Comment (anmenaga) [inline src/System.Management.Automation/engine/hostifaces/PowerShell.cs:5319]: This does not look like it is related to bumping .NET version. Probably better to remove this from this PR.

File: assets/wix/files.wxs [hunk 2]
```diff
@@ -3063,8 +3063,11 @@
       <Component Id="cmp2BB002B115C247FFB3353C80CAAC10B3">
         <File Id="filF1771BA6F8A140FFB77E7558D035A0F2" KeyPath="yes" Source="$(var.ProductSourcePath)\System.IO.Compression.Native.dll" />
       </Component>
-      <Component Id="cmpD9AA61B3F5F44A7B99E535A712E8310D">
-        <File Id="fil16547FB6A5EC4C6791FFFF5D561F2D31" KeyPath="yes" Source="$(var.ProductSourcePath)\mscordaccore_$(var.FileArchitecture)_$(var.FileArchitecture)_6.0.21.10212.dll" />
+      <Component Id="cmp3547A55EA9D04784B4CF2FC717447E16">
+        <File Id="fil463DDA419769417A8D5A34C433E194F2" KeyPath="yes" Source="$(var.ProductSourcePath)\System.Speech.dll" />
```
Comment (anmenaga) [inline assets/wix/files.wxs:3067]: Just curious, why this `System.Speech.dll` is mentioned here?

**Review by anmenaga — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/System.Management.Automation.csproj [hunk 3]
```diff
@@ -16,16 +16,16 @@
...
     <PackageReference Include="Microsoft.ApplicationInsights" Version="2.17.0" />
     <!-- the following package(s) are from https://github.com/dotnet/corefx -->
-    <PackageReference Include="Microsoft.Win32.Registry.AccessControl" Version="6.0.0-preview.1.21102.12" />
-    <PackageReference Include="System.Configuration.ConfigurationManager" Version="6.0.0-preview.1.21102.12" />
-    <PackageReference Include="System.DirectoryServices" Version="6.0.0-preview.1.21102.12" />
-    <PackageReference Include="System.IO.FileSystem.AccessControl" Version="6.0.0-preview.1.21102.12" />
-    <PackageReference Include="System.Management" Version="6.0.0-preview.1.21102.12" />
-    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="6.0.0-preview.1.21102.12" />
-    <PackageReference Include="System.Security.AccessControl" Version="6.0.0-preview.1.21102.12" />
-    <PackageReference Include="System.Security.Cryptography.Pkcs" Version="5.0.1" />
```
Comment (anmenaga) [inline src/System.Management.Automation/System.Management.Automation.csproj:26]: I'm curios why `System.Security.Cryptography.Pkcs` was on version="5.0.1" while all neighbor items were on `6.0.0-preview.1` - is this expected or something we missed previously?

File: assets/wix/files.wxs (hunk 2 above)
Comment (adityapatwardhan) [reply assets/wix/files.wxs:3067]: Comes with the framework i guess

File: src/System.Management.Automation/engine/hostifaces/PowerShell.cs (hunk 1 above)
Comment (adityapatwardhan) [reply src/System.Management.Automation/engine/hostifaces/PowerShell.cs:5319]: This is due to this issue i reported: https://github.com/dotnet/roslyn/issues/51849

File: assets/wix/files.wxs (hunk 2 above)
Comment (iSazonov) [reply assets/wix/files.wxs:3067]: This should be reported to .Net Runtime repo.

File: src/System.Management.Automation/System.Management.Automation.csproj (hunk 3 above)
Comment (iSazonov) [reply src/System.Management.Automation/System.Management.Automation.csproj:26]: I guess not all packages was updated in .Net Runtime for Preview.1.

Comment (adityapatwardhan) [reply src/System.Management.Automation/System.Management.Automation.csproj:26]: It seems we missed updating a couple of packages for preview 1. Should not be a problem this release though.

**Review by rjmholt — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
Snapstore has an outage which is cause the link tests to fail.

**Pull request merged**